src/rules/noObjectComparisonRule.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isBinaryExpression, isTypeFlagSet } from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+import { codeExamples } from "./code-examples/noObjectComparison.examples";
+
+const OPTION_ALLOW_EQUAL = "allow-equal";
+
+interface Options {
+    allowEquals: boolean;
+}
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-object-comparison",
```

Comment:
> `"no-object-comparison"`

That name seems a little inaccurate, since this rule can allow some kinds of object comparisons (and sometimes that is preferable). I'm also not a fan of the `no-` prefix for rule names, as that makes it less convenient to add the opposite behavior as an option and plays with alphabetical ordering of rules. 

How about `object-comparison` as a rule name?

Reply:
I thought in this case the `no-` was ok because all options will only weaken the constraints, but they will never reverse it, so it will never "force" you to do object comparison.

But removing the `no` also works for me.